csrc/ir/nodes.cpp

```diff
@@ -1334,6 +1334,44 @@ std::string ReductionOp::toInlineString(int indent_size) const {
   NVF_CHECK(false, "Tensor op can not be printed inline");
 }
 
+std::vector<PolymorphicValue> ReductionOp::evaluate(
+    const ExpressionEvaluator& ee,
+    const std::vector<PolymorphicValue>& inputs) const {
+  const auto& input = inputs.at(0).as<at::Tensor>();
+  const auto output = out()->as<TensorView>();
+
+  NVF_ERROR(output->getRFactorDomain().size() == 0, 
+    "Evaluation for rFactored reductions is not supported.");
+
+  std::vector<int64_t> reduction_axes;
+  for (const auto i: c10::irange(output->getRootDomain().size())){
+    auto ax = output->getRootDomain().at(i);
+    if (ax->isReduction())
+      reduction_axes.push_back(i);
+  }
+  switch (getReductionOpType()) {
+    case BinaryOpType::Add:
+      return {at::sum(input, reduction_axes)};
+      break;
+    case BinaryOpType::Max:
+      NVF_CHECK(
+        reduction_axes.size() == 1, 
+        "Operator type: ", 
+        getReductionOpType(), 
+        " expects one reduction dimension. Found ",
+        reduction_axes.size(), " reduction dimensions.")
+      return {std::get<0>(at::max(input, reduction_axes[0]))};
```

Comment:
Should we use `at::amax`? https://pytorch.org/docs/stable/generated/torch.amax.html